x-pack/plugins/cases/public/components/custom_fields/form_fields.tsx

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { memo, useCallback, useMemo, useState } from 'react';
+
+import { UseField } from '@kbn/es-ui-shared-plugin/static/forms/hook_form_lib';
+import { Field } from '@kbn/es-ui-shared-plugin/static/forms/components';
+import type { CustomFieldTypes, CustomFieldBuildType } from './types';
+import { customFieldTypesValues } from './schema';
+import { FieldTypeSelector } from './field_type/field_type_selector';
+import { builderMap } from './builder';
+
+interface FormFieldsProps {
+  isSubmitting?: boolean;
+}
+
+const FormFieldsComponent: React.FC<FormFieldsProps> = ({ isSubmitting }) => {
+  const [selectedType, setSelectedType] = useState<CustomFieldTypes>(customFieldTypesValues[0]);
+  const handleTypeChange = useCallback(
+    (val: CustomFieldTypes) => {
+      setSelectedType(val);
+    },
+    [setSelectedType]
+  );
+
+  const builtCustomField: CustomFieldBuildType[] | null = useMemo(() => {
+    if (!customFieldTypesValues) {
+      return null;
+    }
+
+    const builder = builderMap[selectedType];
+
+    if (builder == null) {
+      return null;
+    }
+
+    const customFieldBuilder = builder();
+
+    return customFieldBuilder.build();
+  }, [selectedType]);
+
+  const ConfigurePage = builtCustomField?.length && builtCustomField[0]?.ConfigurePage;
+
+  return (
+    <>
+      <UseField
+        path="fieldLabel"
+        component={Field}
+        componentProps={{
+          euiFieldProps: {
+            'data-test-subj': 'custom-field-label-input',
+            fullWidth: true,
+            autoFocus: true,
+            isLoading: isSubmitting,
+          },
+        }}
+      />
+      <UseField
+        path="fieldType"
+        component={FieldTypeSelector}
```

Comment:
Instead of creating our own components, we can use the `SuperSelectField` or `SelectField` from the form lib. You can use the `onChange` prop to listen to changes in the field type (which field is selected).